app/models/vote.rb

```diff
@@ -0,0 +1,27 @@
+class Vote < ActiveRecord::Base
+  belongs_to :voteable, polymorphic: true
+  validates :user_id, presence: true
+
+  class << self
+    def add_vote(subject, subject_id, user_id)
+      subject_of_vote = subject.find_by(id: subject_id)
+      if subject_of_vote
+        unless voted?(subject, subject_id, user_id)
+          subject_of_vote.votes.create(user_id: user_id)
+          return subject.find_by(id: subject_id).votes.count
+        end
+      end
+      nil
+    end
+
+    def voted?(subject, subject_id, user_id)
+      vote = find_by(user_id: user_id, voteable_type: "#{subject}", voteable_id: subject_id)
+      vote.nil? ? false : true
```

Comment:
Any need for explicitly returning `true` or `false`?

I remembered this, instead of fetching a whole object to check for existence, we could use `:exists?` 
> Vote.exists?(user_id: 1, voteable_type: "type", voteable_id: 1) #=> true or false